### PR TITLE
build snapcraft

### PR DESCRIPTION
**Pull request by coryschwartz**
publish snap to the snapcraft store

### CODE REVIEW

**Comment by coryschwartz** [comment]
I made some changes to the snapchat build aside from adding the CI logic.

Origionally, I was using the pre-built binaries and just copying them into the snap, and was using ubuntu "core18" as the snap base. With this base, the linking was compatible with the golang docker image (which is debian). However, but this turned out to be a headache to get all the linking right with the golang-16.4 docker image. The binaries built in the build-all step have enough incompatibilty with package versions that it was easier to simply re-build it for the snap. 

So now, the binary is built as part of the snap construction and is based on ubuntu 20.04 -- although it will be installable on any linux OS running snap.

**Comment by coryschwartz** [comment]
I made some improvements to make sure this works with strict confinement, and added more detail for https://snapcraft.io/lotus-filecoin

File: .circleci/config.yml [hunk 1]
```diff
@@ -861,3 +889,29 @@ workflows:
...
+          filters:
+            branches:
+              ignore:
+                - /.*/
+            tags:
+              only:
+                - /^v\d+\.\d+\.\d+(-rc\d+)?$/
+      - publish-snapcraft:
+          name: publish-snapcraft-tmp
+          channel: edge
```
Comment (magik6k) [inline .circleci/config.yml:904]: Do we want to keep this?

**Review by magik6k — APPROVED** [review]
Looks good, just one question.

File: .circleci/config.yml (hunk 1 above)
Comment (coryschwartz) [reply .circleci/config.yml:904]: oh, no thanks for catching that.

**Review by magik6k — APPROVED** [review]

**Pull request merged**